bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3251,6 +3268,41 @@ void ReplicaImp::onMessage<SimpleAckMsg>(SimpleAckMsg *msg) {
   delete msg;
 }
 
+template <>
+void ReplicaImp::onMessage<ReplicaRestartReadyMsg>(ReplicaRestartReadyMsg *msg) {
+  LOG_INFO(GL,
+           "Recieved  ReplicaRestartReadyMsg from sender_id " << std::to_string(msg->idOfGeneratedReplica())
+                                                              << " with seq_num" << std::to_string(msg->seqNum()));
+  restart_ready_list_.emplace_back(std::make_unique<ReplicaRestartReadyMsg>(msg));
```

Comment:
We need to check for duplications (to ensure that a Byzantine replica doesn't send multiple readyToRestart messages.
Maybe consider using map instead of a vector

Reply:
Also, before pushing it, we need to validate its signature